cmd/metacache-bucket.go

```diff
@@ -317,20 +313,28 @@ func (b *bucketMetacache) cloneCaches() (map[string]metacache, map[string][]stri
 // Deletes are performed concurrently.
 func (b *bucketMetacache) deleteAll() {
 	ctx := context.Background()
-	ez, ok := newObjectLayerFn().(*erasureServerPools)
+
+	objAPI := newObjectLayerFn()
+	if objAPI == nil {
+		return
+	}
+
+	ez, ok := objAPI.(*erasureServerPools)
```

Comment:
Seems like a lot of extra code for doing the same thing?

Also this function is only used for testing, so no functional change will come.